### PR TITLE
Feat(web): Deprecate Dropdown non flow and modifier placement DS-1132

### PR DESCRIPTION
**Pull request by crishpeen**
Use data attribute to set the placement now.
Use flow placement values, e.g. `top-left` is now `top-start`.

### Issue reference

---

### Before submitting the PR, please make sure you do the following

- [ ] Read the [Contributing Guidelines](https://github.com/lmc-eu/spirit-design-system/blob/main/CONTRIBUTING.md).
- [ ] Follow the [PR Title/Commit Message Convention](https://github.com/lmc-eu/spirit-design-system/blob/main/CONTRIBUTING.md#commit-conventions).
- [ ] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [ ] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

File: packages/web/src/scss/components/Dropdown/README.md [hunk 1]
```diff
@@ -239,14 +249,27 @@
...
+[What are deprecations?][readme-deprecations]
+
+### ⚠️ DEPRECATION NOTICE
+
+CSS modifiers `Dropdown--top`, `Dropdown--rightTop`, `Dropdown--bottom`, etc. are deprecated and will be
+removed in the next major release. Use `data-spirit-placement` attribute instead.
+
+Also both axis side placements are renamed from `top-left`, `top-right`, `right-top`, `right-bottom`,
+etc. to `top-start`, `top-end`, `right-start`, `right-end`, etc. Old names are deprecated and will be
+removed in the next major release.
```
Comment (adamkudrna) [inline packages/web/src/scss/components/Dropdown/README.md:269]:
```suggestion
Also, both axis side placements are renamed from `top-left`, `top-right`, `right-top`, `right-bottom`,
etc. to `top-start`, `top-end`, `right-start`, `right-end`, etc. The old names are deprecated and will be
removed in the next major release.
```

File: packages/web/src/scss/components/Dropdown/README.md [hunk 2]
```diff
@@ -6,14 +6,14 @@
 <div class="DropdownWrapper">
   <button
     data-spirit-toggle="dropdown"
-    data-spirit-target="#dropdownDefault"
+    data-spirit-target="#dropdown-default"
     class="Button Button--primary Button--medium"
     aria-expanded="false"
-    aria-controls="#dropdownDefault"
+    aria-controls="#dropdown-default"
```
Comment (adamkudrna) [inline packages/web/src/scss/components/Dropdown/README.md:12]: `aria-controls`accepts an ID, not a selector:

```suggestion
    aria-controls="dropdown-default"
```

File: packages/web/src/scss/components/Dropdown/README.md [hunk 3]
```diff
@@ -90,15 +90,15 @@
 <div class="DropdownWrapper">
   <button
     data-spirit-toggle="dropdown"
-    data-spirit-target="#dropdownDisabledAutoClose"
+    data-spirit-target="#dropdown-disabled-auto-close"
     class="Button Button--primary Button--medium"
     aria-expanded="false"
-    aria-controls="#dropdownDisabledAutoClose"
+    aria-controls="#dropdown-disabled-auto-close"
     data-spirit-autoclose="true"
```
Comment (adamkudrna) [inline packages/web/src/scss/components/Dropdown/README.md:97]: This doesn't work as expected, reported as https://jira.almacareer.tech/browse/DS-1157.

File: packages/web/src/scss/components/Dropdown/README.md
Comment (adamkudrna) [inline packages/web/src/scss/components/Dropdown/README.md:1]: There is one more occurence in Item README.

**Review by adamkudrna — COMMENTED** [review]
+1 for rewriting the IDs!

**Review by pavelklibani — APPROVED** [review]

**Pull request merged**